web_src/js/components/ActionRunStatus.vue

```diff
@@ -2,31 +2,21 @@
     Please also update the template file above if this vue is modified.
     action status accepted: success, skipped, waiting, blocked, running, failure, cancelled, unknown
 -->
-<script lang="ts">
+<script lang="ts" setup>
 import {SvgIcon} from '../svg.ts';
 
-export default {
-  components: {SvgIcon},
-  props: {
-    status: {
-      type: String,
-      required: true,
-    },
-    size: {
-      type: Number,
-      default: 16,
-    },
-    className: {
-      type: String,
-      default: '',
-    },
-    localeStatus: {
-      type: String,
-      default: '',
-    },
-  },
-};
+withDefaults(defineProps<{
```

Comment:
I guess I would prefer importing these functions explicitly than relying on such "globals":

```ts
import {withDefaults, defineProps} from 'vue';

Reply:
Those are considered as compiler macros throwing: `[[@vue](https://dev.to/vue)/compiler-sfc] defineProps is a compiler macro and no longer needs to be imported.` when imported.

Reply:
Huh, odd decision from Vue authors to encourage what are essential globals. I thought the JS community had moved past such problematic things, but apparently not in the case of Vue.